pkg/asset/machines/clusterapi.go

```diff
@@ -301,10 +304,34 @@ func (c *ClusterAPI) Generate(dependencies asset.Parents) error {
 		mpool.Set(pool.Platform.VSphere)
 
 		platform := ic.VSphere
+		resolver := &net.Resolver{
+			PreferGo: true,
+		}
 
 		for _, v := range platform.VCenters {
-			err := installConfig.VSphere.Networks(ctx, v, platform.FailureDomains)
+			// Defense against potential issues with assisted installer
+			// If the installer is unable to resolve vCenter there is a good possibility
+			// that the installer's install-config has been provided with bogus values.
+
+			ctx, cancel := context.WithTimeout(context.TODO(), 30*time.Second)
```

Comment:
Is 30s enough for both `LookupHost` and `Networks` calls?

Reply:
I would hope so...

```
jcallen@fedora:~/go/src/github.com/openshift/installer$  time host -t A www.google.com 1.1.1.1
Using domain server:
Name: 1.1.1.1
Address: 1.1.1.1#53
Aliases: 

www.google.com has address 142.250.190.36

real    0m0.023s
user    0m0.003s
sys     0m0.005s
```

Reply:
I would still create a new context with timeout for `Networks` since it's creating a session and populating the networks.

Reply:
@r4f4 thanks! I didn't even notice the context being reused. And you are right, Networks since it uses vCenter could be way longer if including resolution.